### PR TITLE
fix(@angular-devkit/schematics-cli): fix `list-schematics` is not printing anything

### LINKED ISSUE #12220: schematics-cli: --list-schematics is broken
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
- [x] schematics
```

### Versions

```
Angular CLI: 6.2.1
Node: 8.11.3
OS: darwin x64
Angular: 
... 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.8.1
@angular-devkit/core         0.8.1
@angular-devkit/schematics   0.8.1
@schematics/angular          0.8.1
@schematics/update           0.8.1
rxjs                         6.2.2
typescript                   2.9.2
```

### Repro steps

1. `npm i -g @angular-devkit/schematics-cli @schematics/angular`
1. Verify that it installed properly via `schematics --help`

These commands give no output at all:
- `schematics @schematics/schematics --list-schematics`
- `schematics @schematics/angular --list-schematics`

### The log given by the failure

No output, even with `--verbose`.

### Desired functionality

The list of schematics in that collection as indicated by `schematics --help`:
```
schematics [CollectionName:]SchematicName [options, ...]

By default, if the collection name is not specified, use the internal collection provided
by the Schematics CLI.

Options:
    --debug             Debug mode. This is true by default if the collection is a relative
                        path (in that case, turn off with --debug=false).
    --allowPrivate      Allow private schematics to be run from the command line. Default to
                        false.
    --dry-run           Do not output anything, but instead just show what actions would be
                        performed. Default to true if debug is also true.
    --force             Force overwriting files that would otherwise be an error.
    --list-schematics   List all schematics from the collection, by name.
    --verbose           Show more information.

    --help              Show this message.

Any additional option is passed to the Schematics depending on
```
This used to work but appears to have been broken at some point. I.e. this is a regression.

### PR DESCRIPTION
**Pull request by alan-agius4**
fix `list-schematics` is not printing anything

Usage:
```
$  schematics @schematics/angular: --list-schematics
$  schematics --list-schematics
```

When no colon is specified, it means that that you are passing a schematic name to be looked up in the default collection.

Closes #12220

### CODE REVIEW

File: packages/angular_devkit/schematics_cli/bin/schematics.ts [hunk 1]
```diff
@@ -36,21 +44,26 @@ function usage(exitCode = 0): never {
     Options:
         --debug             Debug mode. This is true by default if the collection is a relative
                             path (in that case, turn off with --debug=false).
+
```
Comment (alan-agius4) [inline packages/angular_devkit/schematics_cli/bin/schematics.ts]: The console is quite cluttered without break lines

**Comment by alan-agius4** [comment]
@alexeagle I have added a couple of test as previously `schematics-cli` was not testable. However, in order to do that I had to do a couple of other changes.

File: packages/angular_devkit/schematics_cli/bin/schematics.ts [hunk 2]
```diff
@@ -68,192 +45,248 @@ function usage(exitCode = 0): never {
...
+    '--': true,
+  });
+
+  /** Create the DevKit Logger used through the CLI. */
+  const logger = createConsoleLogger(argv['verbose'], stdout, stderr);
+
+  /**
+ * Show usage of the CLI tool, and exit the process.
+ */
+  function usage(exitCode: 1 | 0 = 0): 1 | 0 {
```
Comment (hansl) [inline packages/angular_devkit/schematics_cli/bin/schematics.ts]: Usage should probably be outside of main, which will bring this function under the size limit for the no-big-function lint rule.

Comment (alan-agius4) [reply packages/angular_devkit/schematics_cli/bin/schematics.ts]: @hansl Updated as requested

**Comment by alexeagle** [comment]
Dismiss stale review is the wrong thing to do. It throws away an "approved with the following one change".
I think we have a setting somewhere causing this which we should change...

File: packages/angular_devkit/schematics_cli/bin/schematics.ts [hunk 3]
```diff
@@ -263,11 +264,46 @@ workflow.execute({
...
+      --allowPrivate      Allow private schematics to be run from the command line. Default to
+                          false.
+
+      --dry-run           Do not output anything, but instead just show what actions would be
+                          performed. Default to true if debug is also true.
+
+      --force             Force overwriting files that would otherwise be an error.
+
+      --list-schematics   List all schematics from the collection, by name. A collection name
+                          should be suffixed by a colon. Example: '@schematics/schematics:'.
```
Comment (Splaktar) [inline packages/angular_devkit/schematics_cli/bin/schematics.ts]: Thank you for this example with the colon! It's not something that would be expected, but this example should allow people to figure it out.

**Comment by Splaktar** [comment]
Thank you for jumping on this quickly! I'm planning to demonstrate a bit of this functionality for my schematics talk at AngularMix.

**Pull request merged**